modules/mono/glue/cs_files/Quat.cs

```diff
@@ -178,115 +190,146 @@ public Quat Slerpni(Quat b, real_t t)
 
             return new Quat
             (
-                invFactor * x + newFactor * b.x,
-                invFactor * y + newFactor * b.y,
-                invFactor * z + newFactor * b.z,
-                invFactor * w + newFactor * b.w
+                invFactor * X + newFactor * b.X,
+                invFactor * Y + newFactor * b.Y,
+                invFactor * Z + newFactor * b.Z,
+                invFactor * W + newFactor * b.W
             );
         }
 
         public Vector3 Xform(Vector3 v)
         {
             Quat q = this * v;
             q *= Inverse();
-            return new Vector3(q.x, q.y, q.z);
+            return new Vector3(q.X, q.Y, q.Z);
         }
 
+
+        // Static Readonly Properties
+        public static Quat Identity { get; } = new Quat(0f, 0f, 0f, 1f);
+
         // Constructors 
         public Quat(real_t x, real_t y, real_t z, real_t w)
         {
-            this.x = x;
-            this.y = y;
-            this.z = z;
-            this.w = w;
-        }   
+            X = x;
+            Y = y;
+            Z = z;
+            W = w;
+        }
+
         public Quat(Quat q)
-        {                     
-            x = q.x;
-            y = q.y;
-            z = q.z;
-            w = q.w;
+        {
+            this = q;
         }
-        
+
+        public Quat(Basis basis)
+        {
+            this = basis.Quat();
+        }
+
+        public Quat(Vector3 eulerYXZ)
+        {
+            real_t half_a1 = eulerYXZ.y * (real_t)0.5;
+            real_t half_a2 = eulerYXZ.x * (real_t)0.5;
+            real_t half_a3 = eulerYXZ.z * (real_t)0.5;
+
+            // R = Y(a1).X(a2).Z(a3) convention for Euler angles.
+            // Conversion to quaternion as listed in https://ntrs.nasa.gov/archive/nasa/casi.ntrs.nasa.gov/19770024290.pdf (page A-6)
+            // a3 is the angle of the first rotation, following the notation in this reference.
+
+            real_t cos_a1 = Mathf.Cos(half_a1);
+            real_t sin_a1 = Mathf.Sin(half_a1);
+            real_t cos_a2 = Mathf.Cos(half_a2);
+            real_t sin_a2 = Mathf.Sin(half_a2);
+            real_t cos_a3 = Mathf.Cos(half_a3);
+            real_t sin_a3 = Mathf.Sin(half_a3);
+
+            X = sin_a1 * cos_a2 * sin_a3 + cos_a1 * sin_a2 * cos_a3;
+            Y = sin_a1 * cos_a2 * cos_a3 - cos_a1 * sin_a2 * sin_a3;
+            Z = -sin_a1 * sin_a2 * cos_a3 + cos_a1 * cos_a2 * sin_a3;
+            W = sin_a1 * sin_a2 * sin_a3 + cos_a1 * cos_a2 * cos_a3; SetEuler(eulerYXZ);
```

Comment:
Infinite loop. You have `SetEuler` calling this constructor and this constructor is calling `SetEuler`. Also, change to an overload of `Set` or possibly an `Euler` property.

Reply:
Thanks again, this one was a copy paste error.  Thanks for catching it!